namerd/iface/control-http/src/main/scala/io/buoyant/namerd/iface/HttpControlService.scala

```diff
@@ -397,12 +397,13 @@ class HttpControlService(storage: DtabStore, delegate: Ns => NameInterpreter, na
     }
   }
 
-  private[this] def handleGetDelegate(ns: String, path: Path, extraDtab: String): Future[Response] = {
+  private[this] def handleGetDelegate(ns: String, path: Path, extraDtab: Option[String]): Future[Response] = {
     getDtab(ns).flatMap {
       case Some(dtab) =>
         delegate(ns) match {
           case delegator: Delegator =>
-            DelegateApiHandler.getDelegateRsp((dtab.dtab ++ Dtab.read(extraDtab)).show, path.show, delegator)
+            val fullDtab = extraDtab match { case Some(str) => dtab.dtab ++ Dtab.read(str); case _ => dtab.dtab }
```

Comment:
I'd break up this line:
``
val fullDtab = extraDtab match {
  case Some(str) => ...
  case None => ...
}
```